internal/handlers/common/sort.go

```diff
@@ -14,10 +14,173 @@
 
 package common
 
-import "github.com/FerretDB/FerretDB/internal/types"
+import (
+	"math"
+	"sort"
+	"strings"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// sortType represents sort type for $sort aggregation.
+type sortType int
+
+const (
+	ascending sortType = iota
+	descending
+)
 
 // SortDocuments sorts given documents in place according to the given sorting conditions.
 func SortDocuments(docs []*types.Document, sort *types.Document) error {
-	// TODO
+	if sort.Len() == 0 {
+		return nil
+	}
+
+	if sort.Len() > 32 {
+		return lazyerrors.Errorf("maximum sort keys exceeded: %v", sort.Len())
+	}
+
+	sortFuncs := make([]sortFunc, len(sort.Keys()))
+	for i, sortKey := range sort.Keys() {
+		sortField, err := sort.Get(sortKey)
```

Comment:
That's one place where we can and even should use `must.NotFail(sort.Get(sortKey))`. We iterate over keys, so that document must have that key.

Using panics in "impossible" situations (that's what must.NotFail does) instead of returning errors will make fuzzing easier: https://github.com/FerretDB/FerretDB/blob/7613293be8c353eee7b9ced7954424800b7411a3/internal/handlers/handler_fuzz_test.go#L62-L63

Reply:
Thanks, I'll try to keep that in mind in the future.